templates/CRM/Contact/Form/Edit/TagsAndGroups.tpl

```diff
@@ -12,24 +12,24 @@
   <div class="crm-accordion-header">{$title}</div>
   <div class="crm-accordion-body" id="tagGroup">
 {/if}
-    <table class="form-layout-compressed{if $context EQ 'profile'} crm-profile-tagsandgroups{/if}">
+    <table class="form-layout-compressed{if isset($context) && $context EQ 'profile'} crm-profile-tagsandgroups{/if}">
       <tr>
-        {if !$type || $type eq 'tag'}
+        {if empty($type) || (isset($type) && $type eq 'tag')}
           <td>
             <div class="crm-section tag-section">
-              {if $title}{$form.tag.label}<br>{/if}
+              {if !empty($title)}{$form.tag.label}<br>{/if}
               {$form.tag.html}
             </div>
-            {if $context NEQ 'profile'}
+            {if !isset($context) || (isset($context) && $context NEQ 'profile')}
               {include file="CRM/common/Tagset.tpl"}
             {/if}
           </td>
         {/if}
-        {if !$type || $type eq 'group'}
+        {if isset($type) && $type eq 'group'}
```

Comment:
I don't think that's exactly the same. I think it should be 
```suggestion
        {if empty($type) || $type eq 'group'}
```
Disclaimer: this is making my brain hurt so I could be wrong

Reply:
Markdown table to the rescue!

The integer 0 one is fun.

type | {if !$type \|\| $type eq 'group'} | {if isset($type) &amp;&amp;   $type eq 'group'} | {if empty($type) \|\| $type eq   'group'}
-- | -- | -- | --
unset | TRUE | FALSE | TRUE
null | TRUE | FALSE | TRUE
group | TRUE | TRUE | TRUE
banana | FALSE | FALSE | FALSE
"" | TRUE | FALSE | TRUE
"0" | TRUE | FALSE | TRUE
"1" | FALSE | FALSE | FALSE
0 | TRUE | TRUE | TRUE
1 | FALSE | FALSE | FALSE
TRUE | TRUE | TRUE | TRUE
FALSE | TRUE | FALSE | TRUE

Reply:
LOL, you never know when you expect a group but it's actually a banana.